pkg/kubecost/asset_test.go

```diff
@@ -1479,3 +1479,209 @@ func TestAssetSetRange_MarshalJSON(t *testing.T) {
+	if alloc.Minutes() > 300.0 {
+		t.Errorf("accumulating AllocationSetRange: expected %f or less minutes; actual %f", 300.0, alloc.Minutes())
+	}
+}
+
+func TestAssetSetRange_AccumulateBy_Week(t *testing.T) {
+	ago9d := time.Now().UTC().Truncate(day).Add(-9 * day)
+	ago8d := time.Now().UTC().Truncate(day).Add(-8 * day)
+	ago7d := time.Now().UTC().Truncate(day).Add(-7 * day)
+	ago6d := time.Now().UTC().Truncate(day).Add(-6 * day)
+	ago5d := time.Now().UTC().Truncate(day).Add(-5 * day)
+	ago4d := time.Now().UTC().Truncate(day).Add(-4 * day)
+	ago3d := time.Now().UTC().Truncate(day).Add(-3 * day)
+	ago2d := time.Now().UTC().Truncate(day).Add(-2 * day)
+	yesterday := time.Now().UTC().Truncate(day).Add(-day)
+	today := time.Now().UTC().Truncate(day)
+
+	ago9dAS := GenerateMockAssetSet(ago9d, day)
+	ago8dAS := GenerateMockAssetSet(ago8d, day)
+	ago7dAS := GenerateMockAssetSet(ago7d, day)
+	ago6dAS := GenerateMockAssetSet(ago6d, day)
+	ago5dAS := GenerateMockAssetSet(ago5d, day)
+	ago4dAS := GenerateMockAssetSet(ago4d, day)
+	ago3dAS := GenerateMockAssetSet(ago3d, day)
+	ago2dAS := GenerateMockAssetSet(ago2d, day)
+	yesterdayAS := GenerateMockAssetSet(yesterday, day)
+	todayAS := GenerateMockAssetSet(today, day)
+
+	asr := NewAssetSetRange(ago9dAS, ago8dAS, ago7dAS, ago6dAS, ago5dAS, ago4dAS, ago3dAS, ago2dAS, yesterdayAS, todayAS)
+	asr, err := asr.Accumulate(AccumulateOptionWeek)
+	if err != nil {
+		t.Fatalf("unexpected error calling accumulateBy: %s", err)
+	}
+
+	if len(asr.Assets) != 2 && len(asr.Assets) != 3 {
+		t.Fatalf("expected 2 or 3 asset sets, got:%d", len(asr.Assets))
+	}
+
+	for _, as := range asr.Assets {
+		if as.Window.Duration() < time.Hour*24 || as.Window.Duration() > time.Hour*24*7 {
+			t.Fatalf("expected window duration to be between 1 and 7 days, got:%s", as.Window.Duration().String())
+		}
+	}
+}
+
+func TestAssetSetRange_AccumulateBy_Month(t *testing.T) {
+	prevMonth1stDay := time.Date(2020, 01, 29, 0, 0, 0, 0, time.UTC)
+	prevMonth2ndDay := time.Date(2020, 01, 30, 0, 0, 0, 0, time.UTC)
+	prevMonth3ndDay := time.Date(2020, 01, 31, 0, 0, 0, 0, time.UTC)
+	nextMonth1stDay := time.Date(2020, 02, 01, 0, 0, 0, 0, time.UTC)
+
+	prev1AS := GenerateMockAssetSet(prevMonth1stDay, day)
+	prev2AS := GenerateMockAssetSet(prevMonth2ndDay, day)
+	prev3AS := GenerateMockAssetSet(prevMonth3ndDay, day)
+	nextAS := GenerateMockAssetSet(nextMonth1stDay, day)
+
+	asr := NewAssetSetRange(prev1AS, prev2AS, prev3AS, nextAS)
+	asr, err := asr.Accumulate(AccumulateOptionMonth)
+	if err != nil {
+		t.Fatalf("unexpected error calling accumulateBy: %s", err)
+	}
+
+	if len(asr.Assets) != 2 {
+		t.Fatalf("expected 2 assets sets, got:%d", len(asr.Assets))
+	}
+
+	for _, as := range asr.Assets {
+		if as.Window.Duration() < time.Hour*24 || as.Window.Duration() > time.Hour*24*31 {
+			t.Fatalf("expected window duration to be between 1 and 7 days, got:%s", as.Window.Duration().String())
```

Comment:
@saweber would this reflect your condition? 
```suggestion
			t.Fatalf("expected window duration to be between 1 and 31 days, got:%s", as.Window.Duration().String())
```